### PR TITLE
[SYCL][DOC][Ordered Queue] Queue Property Proposal

### PR DESCRIPTION
**Pull request by jbrodman**
New proposal meant to replace Ordered Queue.

### CODE REVIEW

File: sycl/doc/extensions/OrderedQueue/OrderedQueue_v2.adoc [hunk 1]
```diff
@@ -0,0 +1,31 @@
+= SYCL Proposals: Queue Order Properties
+James Brodman <james.brodman@intel.com>
+v0.2
+:source-highlighter: pygments
+:icons: font
+== Introduction
+This document presents an addition proposed for a future version of the SYCL Specification.  The goal of this proposal is to reduce the complexity and verbosity of using SYCL for programmers.
+
+== Queue Order Properties
+Queues in SYCL are out-of-order by default.  SYCL constructs directed acyclic graphs to ensure tasks are properly ordered based on their data dependences.  However, many programs only require linear DAGs.  The overheads of constructing and managing DAGs are unnecessary for this class of program.  The `in_order` queue property is proposed to serve this class of programs by providing programmer-specified in-order semantics.  This property is used with the existing `property_list` mechanism that is part of the normal SYCL `queue`.
```
Comment (Pennycook) [inline sycl/doc/extensions/OrderedQueue/OrderedQueue_v2.adoc]: Should "task" be "kernel" here?

File: sycl/doc/extensions/OrderedQueue/OrderedQueue_v2.adoc [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+== Queue Order Properties
+Queues in SYCL are out-of-order by default.  SYCL constructs directed acyclic graphs to ensure tasks are properly ordered based on their data dependences.  However, many programs only require linear DAGs.  The overheads of constructing and managing DAGs are unnecessary for this class of program.  The `in_order` queue property is proposed to serve this class of programs by providing programmer-specified in-order semantics.  This property is used with the existing `property_list` mechanism that is part of the normal SYCL `queue`.  
+
+.Proposed Queue Property
+[cols="^50,50",options="header"]
+|===
+
+|Property |Description
+|`property::queue::in_order`
+| The `in_order` property adds the requirement that the SYCL queue provides in-order semantics where tasks are executed in the order in which they are submitted. Tasks submitted in this fashion can be viewed as having an implicit dependence on the previously submitted operation.
```
Comment (Pennycook) [inline sycl/doc/extensions/OrderedQueue/OrderedQueue_v2.adoc]: Same question as above.

File: sycl/doc/extensions/OrderedQueue/OrderedQueue_v2.adoc (hunk 1 above)
Comment (jbrodman) [reply sycl/doc/extensions/OrderedQueue/OrderedQueue_v2.adoc]: No - because the operation could be a copy, etc.

Comment (Pennycook) [reply sycl/doc/extensions/OrderedQueue/OrderedQueue_v2.adoc]: Ah, I see.  Is there any other word in the specification that describes the collection of things that can appear in command group scope?  If not, "task" is fine.

**Comment by jbrodman** [comment]
@bader Is this a legitimate failure?

**Comment by bader** [comment]
Do you mean this one: http://ci.llvm.intel.com:8010/#/builders/34/builds/139/steps/11/logs/stdio?
No idea. @tfzhu, @vladimirlaz do you know what is going on here?

@Pennycook, @jbrodman, please, resolve the comments from John's review if you think this PR is ready to be merged.

File: sycl/doc/extensions/OrderedQueue/OrderedQueue_v2.adoc (hunk 1 above)
Comment (Pennycook) [reply sycl/doc/extensions/OrderedQueue/OrderedQueue_v2.adoc]: I found that the specification describes the behavior of `wait` as "Performs a blocking wait for the completion of all enqueued tasks in the queue.", so what you have here isn't inconsistent with the specification.

I'd still prefer that there was one word used consistently everywhere to capture this concept, but the issue is larger than this PR and we should revisit it later.

**Comment by jbrodman** [comment]
Seems better now.  Rebased just in case.

**Pull request merged**